### PR TITLE
SpellSystem and misc opcodes

### PR DESCRIPTION
**Pull request by Zyres**
Overall:
- SpellInfo is now a class (basically the old SpellEntry structure on wotlk)
- Moved some functions from player class to SpellInfo (to request information for a spell from the player class looks awkward)
- Added some helper functions (cleanup some script parts)
- Misc packet updates

Logging:
- Introduced logflags

### CODE REVIEW

**Comment by DankoDJ** [comment]
@Zyres o.O

File: src/world/Units/Unit.cpp [hunk 1]
```diff
@@ -3196,7 +3196,7 @@ void Unit::Strike(Unit* pVictim, uint32 weapon_damage_type, SpellInfo* ability,
         {
             case MELEE:   // melee main hand weapon
                 it = disarmed ? NULL : pr->GetItemInterface()->GetInventoryItem(EQUIPMENT_SLOT_MAINHAND);
-                self_skill = float2int32(pr->CalcRating(PLAYER_FIELD_COMBAT_RATING_1 + PCR_MELEE_MAIN_HAND_SKILL));
+                self_skill = float2int32(pr->CalcRating(PLAYER_FIELD_COMBAT_RATING_1 + PCR_MELEE_MAIN_HAND_SKILL)) + 375;
```
Comment (Zyres) [inline src/world/Units/Unit.cpp]: Not sure with these values at all.

**Comment by Zyres** [comment]
I need a proper way to controle the log....
--- a few commits for this ---

**Comment by Zyres** [comment]
Done with log for now
@DankoDJ -done

**Comment by DankoDJ** [comment]
We should use the Color implemented by @armm77 again.
oLog::Map is not in use anymore, remove this function.

**Comment by Zyres** [comment]
color

**Pull request merged**